src/main/scala/outwatch/dom/DomTypes.scala

```diff
@@ -0,0 +1,125 @@
+package outwatch.dom
+
+import com.raquo.domtypes.generic.builders._
+import com.raquo.domtypes.generic.keys
+import com.raquo.domtypes.generic.codecs._
+import com.raquo.domtypes.generic.defs.attrs._
+import com.raquo.domtypes.generic.defs.reflectedAttrs._
+import com.raquo.domtypes.generic.defs.props._
+import com.raquo.domtypes.generic.defs.styles._
+import com.raquo.domtypes.generic.defs.sameRefTags._
+import com.raquo.domtypes.jsdom.defs.eventProps._
+import cats.effect.IO
+import org.scalajs.dom
+import helpers._
+
+trait DomTypes
+  extends DomTags
+  with DomAttrs
+  with DomProps
+  with DomEvents
+  with DomStyles
+
+private[outwatch] object typehelper {
+  type GenericVNode[T] = VNode_
+}
+import typehelper._
+
+trait VNodeBuilder extends TagBuilder[GenericVNode, VNode_] {
+  // we can ignore information about void tags here, because snabbdom handles this automatically for us based on the tagname.
+  override def tag[Ref <: VNode_](tagName: String, void: Boolean): VNode_ =
+    VTree(tagName, Seq.empty)
+}
+
+trait DomTags
+  extends DocumentTags[GenericVNode, VNode_]
+  with GroupingTags[GenericVNode, VNode_]
+  with TextTags[GenericVNode, VNode_]
+  with FormTags[GenericVNode, VNode_]
+  with SectionTags[GenericVNode, VNode_]
+  with EmbedTags[GenericVNode, VNode_]
+  with TableTags[GenericVNode, VNode_]
+  with VNodeBuilder
+object DomTags extends DomTags
+
+trait DomTagsExtra
+  extends MiscTags[GenericVNode, VNode_]
+  with VNodeBuilder
+object DomTagsExtra extends DomTagsExtra
+
+object CodecBuilder {
+  type Attribute[T, _] = AttributeBuilder[T]
+  type Property[T, _] = PropertyBuilder[T]
+
+  def encodeProperty[V](codec: Codec[V, _]): V => String =
+    // codec.encode may encode to null values
+    (codec.encode _) andThen (v => if (v != null) v.toString else null)
+
+  def encodeAttribute[V](codec: Codec[V, String]): V => Attribute.Value = codec match {
+    //The BooleanAsAttrPresenceCodec does not play well with snabbdom. it
+    //encodes true as "" and false as null, whereas snabbdom needs true/false
+    //of type boolean (not string) for toggling the presence of the attribute.
+    case _: BooleanAsAttrPresenceCodec.type => (v: V) => v.asInstanceOf[Boolean] //TODO
+    case _ => codec.encode _
+  }
+}
+
+trait DomAttrs
+  extends Attrs[AttributeBuilder]
+  with AttrBuilder[AttributeBuilder]
+  with ReflectedAttrs[CodecBuilder.Attribute]
+  with ReflectedAttrBuilder[CodecBuilder.Attribute] {
+
+  override def attr[V](key: String, codec: Codec[V, String]): AttributeBuilder[V] =
+    new AttributeBuilder(key, codec.encode _)
+
+  @inline def reflectedAttr[V, DomPropV](
+    attrKey: String,
+    propKey: String,
+    attrCodec: Codec[V, String],
+    propCodec: Codec[V, DomPropV]
+  ): AttributeBuilder[V] =
+    new AttributeBuilder(attrKey, CodecBuilder.encodeAttribute(attrCodec))
+    //or: new PropertyBuilder(propKey, CodecBuilder.encodeProperty(propCodec))
```

Comment:
Here, we can decide whether we want to represent the reflected attributes as attributes or properties. See https://github.com/raquo/scala-dom-types#reflected-attributes